test/scaladoc/run/t8755/ploogin_1.scala

```diff
@@ -0,0 +1,31 @@
+
+package t8755
+
+import scala.tools.nsc.{Global, Phase}
+import scala.tools.nsc.plugins.{Plugin, PluginComponent}
+import scala.reflect.io.Path
+import scala.reflect.io.File
+
+/** A test plugin.  */
+class Ploogin(val global: Global) extends Plugin {
+  import global._
+
+  val name = "ploogin"
+  val description = "A sample plugin for testing."
+  val components = List[PluginComponent](TestComponent)
+
+  private object TestComponent extends PluginComponent {
+    val global: Ploogin.this.global.type = Ploogin.this.global
+    //override val runsBefore = List("jvm")
```

Comment:
Is the commented out form intentional? Or did you want to remove the line entirely?

Reply:
The three tests are combinations of runsBefore and runsAfter. Which is the override? I need a model to copy. In fact, I commented that I didn't even get that right on the third test and need a test generator.

The other question is: why is runsBefore deemed a second class constraint that has an empty default?